src/main/java/nl/hsac/fitnesse/fixture/slim/web/NgBrowserTest.java

```diff
@@ -1,301 +1,312 @@
+
+import java.lang.reflect.Method;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class NgBrowserTest extends BrowserTest {
+    private final static Set<String> METHODS_NO_WAIT;
+    private Set<String> methodsAnnotatedWithNgNoWait = null;
+    private String angularRoot = "[ng-app], [ng_app], [data-ng-app], [x-ng-app], [ng\\:app]";
+    
+    static {
+    	METHODS_NO_WAIT = ReflectionHelper.validateMethodNames(
+	            NgBrowserTest.class,
+	            "open",
+	            "takeScreenshot",
+	            "location",
+	            "back",
+	            "forward",
+	            "refresh",
+	            "alertText",
+	            "confirmAlert",
+	            "dismissAlert",
+	            "openInNewTab",
+	            "ensureActiveTabIsNotClosed",
+	            "currentTabIndex",
+	            "tabCount",
+	            "ensureOnlyOneTab",
+	            "closeTab",
+	            "setAngularRoot",
+	            "switchToNextTab",
+	            "switchToPreviousTab",
+	            "switchToDefaultContent",
+	            "switchToFrame",
+	            "switchToParentFrame",
+	            "waitForAngularRequestsToFinish",
+	            "secondsBeforeTimeout",
+	            "secondsBeforePageLoadTimeout",
+	            "waitForPage",
+	            "waitForTagWithText",
+	            "waitForClassWithText",
+	            "waitForClass",
+	            "waitForVisible",
+	            "waitSeconds",
+	            "waitMilliseconds",
+	            "waitMilliSecondAfterScroll",
+	            "screenshotBaseDirectory",
+	            "screenshotShowHeight",
+	            "setBrowserWidth",
+	            "setBrowserHeight",
+	            "setBrowserSizeToBy",
+	            "setBrowserSizeToMaximum",
+	            "setGlobalValueTo",
+	            "isImplicitWaitForAngularEnabled",
+	            "setImplicitWaitForAngularTo",
+	            "globalValue",
+	            "clearSearchContext",
+	            "setAngularRoot",
+	            "getAngularRoot");
+    }
+
+    /**
+     * Creates new.
+     */
+    public NgBrowserTest() {
+        setImplicitWaitForAngularTo(true);
+    }
+
+    @Override
+    protected void waitForAngularIfNeeded(Method method) {
+        if (isImplicitWaitForAngularEnabled()
+                && requiresWaitForAngular(method)) {
+            waitForAngularRequestsToFinish();
+        }
+    }
+
+    /**
+     * Determines whether method requires waiting for all Angular requests to finish
+     * before it is invoked.
+     * @param method method to be invoked.
+     * @return true, if waiting for Angular is required, false otherwise.
+     */
+    protected boolean requiresWaitForAngular(Method method) {
+        String methodName = method.getName();
+        return !METHODS_NO_WAIT.contains(methodName) && !hasMethodAnnotatedWithNgNoWait(methodName);
+    }
+
+    private boolean hasMethodAnnotatedWithNgNoWait(String methodName) {

```

Comment:
Make this method work on the method (not its name) and check whether that has the annotation (no need to loop over all methods). And rename the method to `hasNoWaitAnnotation`